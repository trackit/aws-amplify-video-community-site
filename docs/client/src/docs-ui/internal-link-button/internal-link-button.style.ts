import {css} from "emotion";
import {MQFablet} from "../../amplify-ui/styles/media";

export const hostStyle = css`
  display: block;
  margin-bottom: 1.5rem;
`;

export const containerStyle = css`
  a {
    display: flex;
    flex-direction: row;
    text-align: center;
    justify-content: center;
    align-items: center;
    background-color: var(--color-orange-hv);
    padding: 0.675rem;
    border-radius: 0.25rem;
    color: var(--color-white);
    font-weight: 700;
    transition: all 0.6s cubic-bezier(0.165, 0.84, 0.44, 1);
    box-shadow: 0 0.125rem 0.25rem rgba(0, 0, 0, 0.25);
    appearance: none;

    ${MQFablet} {
      max-width: 18rem;
    }

    &:hover {
      text-decoration: none;
      box-shadow: 0 0.125rem 0.25rem rgba(0, 0, 0, 0.5);
    }
  }
`;

export const graphicStyle = css`
  position: relative;
  top: 0.125rem;
  margin-right: 0.5rem;

  img {
    max-width: 1rem;
    max-height: 1rem;
  }
`;
