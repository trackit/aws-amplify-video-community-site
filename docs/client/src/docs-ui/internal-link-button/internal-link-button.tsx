import {h, Component, Host, Prop} from "@stencil/core";
import {
  containerStyle,
  graphicStyle,
  hostStyle,
} from "./internal-link-button.style";

@Component({tag: "docs-internal-link-button", shadow: false})
export class DocsInternalLinkButton {
  /*** the path to redirect to (internal only!) */
  @Prop() readonly href?: string;
  /*** query string parameters to attach to the link */
  @Prop() readonly QSPs: string = "";

  render() {
    return (
      <Host class={hostStyle}>
        <docs-internal-link
          class={containerStyle}
          href={this.href}
          QSPs={this.QSPs}
        >
          <div class={graphicStyle}>
            <slot name="graphic" />
          </div>
          <div>
            <slot name="text" />
          </div>
        </docs-internal-link>
      </Host>
    );
  }
}
