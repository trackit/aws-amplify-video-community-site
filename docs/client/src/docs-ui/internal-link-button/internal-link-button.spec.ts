import {DocsInternalLinkButton} from "./internal-link-button";
import {newSpecPage} from "@stencil/core/testing";

describe("docs-internal-link-button", () => {
  describe("Render logic", () => {
    it("should render", async () => {
      expect(
        (
          await newSpecPage({
            components: [DocsInternalLinkButton],
            html: `<docs-internal-link-button />`,
          })
        ).root,
      ).toMatchSnapshot();
    });
  });
});
