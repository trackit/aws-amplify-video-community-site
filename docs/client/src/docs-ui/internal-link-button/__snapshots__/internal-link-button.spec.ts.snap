// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`docs-internal-link-button Render logic should render 1`] = `
<docs-internal-link-button class="css-fx7q5a">
  <!---->
  <docs-internal-link class="css-sj53un" qsps="">
    <div class="css-8odve0"></div>
    <div></div>
  </docs-internal-link>
</docs-internal-link-button>
`;
